src/main/java/org/spongepowered/api/tag/TagType.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.tag;
+
+import org.spongepowered.api.registry.DefaultedRegistryType;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+@CatalogedBy(TagTypes.class)
+public interface TagType<T extends Taggable> {
+
+    /**
+     * The directory where the tag is stored,
+     * (data/namespace/xxx/(name).json
+     * @see <a href="https://minecraft.fandom.com/wiki/Tag#Folder_structure">https://minecraft.fandom.com/wiki/Tag#Folder_structure</a>
+     * @return Directory name
+     */
+    String datapackDirectory();
```

Comment:
If we're going to have this (and I'm not sure if we necessarily should because an alternative impl might not want to use a flat file structure), this should be a `Path`.

Why is this useful anyway?

Reply:
We need it in Impl but I also don't see a use for it in API.

Reply:
It should be removed then.

Reply:
Agreed, remove it. In fact, the rest of the API for data pack content so far gives no assumption that we're writing anything IO wise.

Reply:
Its needed in the api if you wanted to add your own type of tag, but there's no reason it has to be the name of the directory, I should rename to something like id or name.

Reply:
I've changed it to `id`. Most likely I also want to add a builder for TagType soon so plugins can more easily add a new type of tag.